test/e2e/storage/testsuites/multivolume.go

```diff
@@ -0,0 +1,312 @@
+
+		By("Checking if the 2nd volume exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod, l.resource2.pattern.VolMode, "/mnt/volume2")
+
+		By("Checking if read/write to the 2nd volume works properly")
+		utils.CheckReadWriteToPath(pod, l.resource2.pattern.VolMode, "/mnt/volume2")
+	}
+
+	testAttachOneVolumeToTwoPods := func(l local, requiresSameNode bool) {
+		node := framework.NodeSelection{Name: l.config.ClientNodeName}
+		By(fmt.Sprintf("Creating 1st pod with a volume on %+v", node))
+		pod1, err := framework.CreateSecPodWithNodeSelection(l.cs, l.ns.Name,
+			[]*v1.PersistentVolumeClaim{l.resource1.pvc},
+			false, "", false, false, framework.SELinuxLabel,
+			nil, node, framework.PodStartTimeout)
+		defer func() {
+			framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod1))
+		}()
+		Expect(err).NotTo(HaveOccurred())
+		pod1, err = l.cs.CoreV1().Pods(pod1.Namespace).Get(pod1.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "get pod1")
+		actualNodeName := pod1.Spec.NodeName
+
+		// Set affinity depending on requiresSameNode
+		if requiresSameNode {
+			framework.SetAffinity(&node, actualNodeName)
+		} else {
+			framework.SetAntiAffinity(&node, actualNodeName)
+		}
+
+		By(fmt.Sprintf("Creating 2nd pod with a volume on %+v", node))
+		pod2, err := framework.CreateSecPodWithNodeSelection(l.cs, l.ns.Name,
+			[]*v1.PersistentVolumeClaim{l.resource2.pvc},
+			false, "", false, false, framework.SELinuxLabel,
+			nil, node, framework.PodStartTimeout)
+		defer func() {
+			framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod2))
+		}()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Checking if the volume in 1st pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if read/write to the volume in 1st pod works properly")
+		utils.CheckReadWriteToPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if the volume in 2nd pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod2, l.resource2.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if read/write to the volume in 2nd pod works properly")
+		utils.CheckReadWriteToPath(pod2, l.resource2.pattern.VolMode, "/mnt/volume1")
+
+		// Delete pod2
+		framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod2))
+
+		// Test read/write work well in pod1 after pod2 is deleted
+		By("Rechecking if the volume in 1st pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Rechecking if read/write to the volume in 1st pod works properly")
+		utils.CheckReadWriteToPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+	}
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- same volume mode
+	//   [volume1]  [volume2]
+	It("should attach two volumes with the same volume mode to the same pod", func() {
```

Comment:
There's already a test case that does this [here](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/test/e2e/storage/persistent_volumes.go#L309) and more generically (can handle more than 2). Maybe that should be adapted to the into the test suite.